File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        num_zeros = 0
        total_mult = 1
        for n in nums:
            if n == 0:
                num_zeros += 1

        ret = []

        if num_zeros > 1:
            for i in range(len(nums)):
                ret.append(0)
            return ret

        if num_zeros == 1:
            for n  in nums:
                if n != 0:
                    total_mult *= n
            
            for i, v in enumerate(nums):
                if v == 0:
                    ret.append(total_mult)
                else:
                    ret.append(0)

            return ret

        if num_zeros == 0:
            for n  in nums:
                    total_mult *= n
        
            for i, v in enumerate(nums):
                ret.append(int(total_mult / nums[i]))

        return ret

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int[] ret = new int[2];
        
        for (int i = 0; i < numbers.length; i++) {
            for (int j = 0; j < numbers.length; j++) {
                if (i == j) {
                    continue;
                }

                if (numbers[i] != numbers[j] && numbers[i] + numbers[j] == target) {
                    if (numbers[i] < numbers[j]) {
                        ret[0] =  i + 1;
                        ret[1] =  j + 1;
                    } else {
                        ret[0] = j + 1;
                        ret[1] = i + 1;
                    }
                    break;
                }
            }
        }

        return ret;

    }
}
